### ISSUE TITLE
Improve navigation and discoverability

### ISSUE THREAD

**Issue by paolodamico** [PostHog/posthog#1775]
## Is your feature request related to a problem?

Through numerous customer interviews we now know there seems to be a bit of an issue with our current navigation, in which a) users find it a bit hard to reach the features they need or b) users are unaware of other features/capabilities of PostHog. This [design proposal](https://www.figma.com/file/gtW3o3VKYvf04ws8cZM3Nr/PH-Retention-optimisations?node-id=635%3A28) attempts to tackle that. Also attempting to balance the need to make all features easily discoverable to the problem of having so much stuff that it's overwhelming for new users.

Aside from the above, this design also introduces full support for multiple projects and multiple organizations.

**Release plan**. As this is a particularly intensive change (from a UX perspective), I'm thinking that we can release in the following manner:
1. Multiple feedback rounds (this thread).
1. Initial implementation (very intense UX change, but fortunately it will likely not require a huge code change).
1. Beta testing with select power users (do usability tests and get qualitative feedback from them).
1. Phased release with feature flags (quantitative impact).
1. Full release.

**Comment by jamesefhawkins** [comment]
Apologies for the slow feedback. Mini retro - there is a *lot* going on here, I should have just chunked my thoughts to get some out faster. Previously, I also wouldn't have anticipated you getting this done as I hadn't seen it as a part of your role to design the product this much, so I think I was reticent to put the time into QA'ing this properly due to concerns it'd never be implemented. I should have spoken up!

# First impressions

I like the style. The larger graphics and reduced text in the navigation feels simpler and more focussed on the tool's functionality versus wordy-tiered menus that require more mental effort to process.

## Palette

Whilst the sidebar feels on brand, the greys used differ and somehow feel a little softer than our typical style:
<img width="337" alt="Screenshot 2020-10-10 at 00 45 16" src="https://user-images.githubusercontent.com/47497682/95637373-e1d06a00-0a91-11eb-88c8-96c9baf78a6a.png">

Could we fit better with the [range of colors here](https://posthog.com/handbook/company/branding)? (I'm not convinced the existing palette will suit the app well here, but we could add to this more empathetically)

<img width="569" alt="Screenshot 2020-10-10 at 00 50 13" src="https://user-images.githubusercontent.com/47497682/95637653-910d4100-0a92-11eb-8acf-99f540160932.png">

## Fonts

These are different to the website and brand.  Is there any reason not to use Gosha Sans for the headers?

If so, I'd push for Helvetica Neue given that's used for smaller headers on the website.

## Icons

1 - the plane for retention seems particularly weird
2 - feature flags as a magic wand I think is also a bit weird.
3 - toolbar icon looks a little fussy, but not a big deal at all though

I think overall, the icons are a little bland but are passable - I suspect Lottie could work magic here.

## Language

You've switched from 'People' to 'Persons'. I just really appreciated this change even though it's tiny, given that persons is better for a precise set of things.

# Functionality

## Actions and events and tags

The objective of this redesign is to increase retention by decreasing actions complexity. Here, I feel we have increased complexity by having actions, events AND tags.

> * I believe these are the 3 key benefits of actions that we should be pushing: a) allows grouping multiple events (and of different types) with complex queries; b) helps clean and standardize data (fix typos, if the product or data changes, ...); and c) it works retroactively (huge aha moment, from a customer call).

Above is from #1755. I agree strongly with the  benefits of actions here and that they're something we must retain, but I think *all* of these would be achieved through events with tags, as follows:

* allows grouping of multiple events - if multiple events have the same tag, they are grouped
* helps clean and standardize data - tags can achieve this as well as actions
* it works retroactively - if a tag comes from ie clicking on an element, then it would also apply retroactively to achieve this moment

I think this then frees you to call them 'events' which is an industry standard term. I would remove 'Live' from the sidebar and would replace with 'Events', with it as a list like:

Some_event | tag 1, tag 2, tag 3
Some_other_event | tag 2, tag 5

## Tags layout

<img width="961" alt="Screenshot 2020-10-10 at 00 27 09" src="https://user-images.githubusercontent.com/47497682/95636416-6077d800-0a8f-11eb-9a71-41ba0bf795f1.png">

The instructions struck me as 'outsized' compared to the other places we give instructions. ie perhaps a small icon next to the header on every page requiring instructions would be more consistent. The explanation itself I think is good.

I would envisage the content of this page to be a list of all tags being used in practise and some ability to create a new tag which looks like it hasn't be prototyped yet?

## Org / profile menu

I think it's a great thing to encourage people to carry one profile with them for a variety of reasons that I don't think are fruitful to debate here. I like the work to make the UX support this. There are a couple of things I'd change:

<img width="224" alt="Screenshot 2020-10-10 at 01 01 13" src="https://user-images.githubusercontent.com/47497682/95638203-1b09d980-0a94-11eb-93a0-507aedf687dc.png">

The user profile has a company 'PostHog, Inc' listed *and* two organizations. This is so you can switch organizations. I therefore think it makes more sense to have the organization as the "main" thing showing, then the username underneath. If the username is big, as you currently have, with small text as the organization name, it implies the organization is fixed, which it isn't.

I'd additionally add the global email used by the user added under their username.
<img width="540" alt="Screenshot 2020-10-10 at 01 11 58" src="https://user-images.githubusercontent.com/47497682/95638843-64f3bf00-0a96-11eb-877c-5e7b2867fec8.png">

The profile menu struck me as a little messy, but easy to solve:

* greyed out email above logout can go if you put this under the username
* we have two navigation elements but one as a button 'Organization settings' and one as a link 'Manage billing'. This would make me unclear if billing sits at org level or not (which it should). I'd suggest we remove the billing link entirely and this would sit in the organization's settings page.

**Comment by lottiecoxon** [comment]
#### This is some great feedback 

First: I agree that the font's on the dashboard should share some Gosha Sans and Helvetica Neue !

Second: For the icons, we have three options, the first being the general icon aesthetic, e.g. the simple line vectors that have only one colour. The second option would be this type of icon-

<img width="117" alt="Screenshot 2020-10-12 at 09 52 17" src="https://user-images.githubusercontent.com/65415371/95732709-a3dd6c80-0c70-11eb-8b1a-93ff7802de78.png">

<img width="119" alt="Screenshot 2020-10-12 at 09 52 14" src="https://user-images.githubusercontent.com/65415371/95732697-a0e27c00-0c70-11eb-9882-55e1df4c1563.png">
 
Or the last option, following the route of these icons on the careers page- 

<img width="141" alt="Screenshot 2020-10-12 at 09 54 14" src="https://user-images.githubusercontent.com/65415371/95732915-e7d07180-0c70-11eb-8696-db392b25fc96.png">

My only worry with the more colourful options would be that these could be distracting and make the page *too* loud. 


Concerning the tag alterations in the sidebar I may need to discuss with you about technical side of what this means as I am unsure and don't want to edit things without knowing what they do. But I agree with the concept of taking out some of the side panel options. 


The instructions I am utterly confused by as I actually can't pin point where they are (that is probably me being a little tech blind) 

And finally the profile menu I can change asap !

**Comment by lottiecoxon** [comment]
Are these the instructions you were talking about? @jamesefhawkins 
<img width="382" alt="Screenshot 2020-10-12 at 11 04 54" src="https://user-images.githubusercontent.com/65415371/95739751-dbe9ad00-0c7a-11eb-8306-b2761f51deff.png">

**Comment by lottiecoxon** [comment]
Also what size is this dashboard supposed to be? 
Comparing it to a generic desktop size its looking very small - and therefore all the text looks massive even though its 12-16 px in size- I assume the dashboard should be for desktop sizes not this smaller one? 
<img width="508" alt="Screenshot 2020-10-12 at 11 17 34" src="https://user-images.githubusercontent.com/65415371/95740778-a645c380-0c7c-11eb-94fc-2d5c79d6e675.png">

**Comment by mariusandra** [comment]
Hey @lottiecoxon , apologies if I'm out the loop and didn't see some non-github discussions... but this issue was about the sidebar and other navigation improvements, not dashboards? I might be missing something :)

Dashboard colors seem to be discussed here: https://github.com/PostHog/posthog/issues/1763#issuecomment-705465171

**Comment by lottiecoxon** [comment]
Oh opps - I assumed because James had included screenshots from the dashboard redesign I assumed it was what we were discussing! Happy to move my comments over to the other ticket, but some of the comments are related to icons, type and layout so I shall keep those here for now

**Comment by paolodamico** [comment]
Thanks for the feedback @jamesefhawkins, super helpful! As discussed, I'll address these changes (the profile bar, fonts, colors, language, ...) after #1755 (I'll keep everyone posted here), as we need to ship only one of these changes at a time to reliably measure the impact, and the actions stuff is more likely to have the biggest impact.

@lottiecoxon regarding dashboard colors I agree with @mariusandra that it's better to keep the conversation in #1763. As for the icons, I like the styling of the new careers page but they should definitely be monochromatic. We can use a combination of library icons and custom made (especially for features like retention, actions or feature flags that it's harder to find a well-suited already existing icon). It'd be super useful to get your help on the icons and the tags/actions example section.

Finally regarding the size of the canvases don't worry about it, the screens are indeed meant for regular desktop sizes but I find it easier to design with larger constraints, so basically they'll just resize appropriately based on our current font sizes.

**Comment by Twixes** [comment]
My honest gripe with this design: it doesn't exactly look coherent and modern to me. The shadows are a bit random, rounding applied liberally, overall I don't feel like elements fit each other. Again, just my opinion, but I'd shoot for something cleaner (linked [my proposal](https://www.figma.com/file/irDn9pKCYOrF3WHf0liOmH/PostHog-Navigation-Redesign-Alt?node-id=1%3A14) on Slack before, just as possible inspiration).

**Comment by paolodamico** [comment]
Thanks for the feedback @Twixes, definitely fair. Honestly this is a more about the UX impact than the UI, and I think that's where the most positive impact for this will come from. The shadows are not indeed standardized because to my knowledge we don't have a standard yet (other than Ant's, which we'd use for the implementation). To be honest I'm not a fan of this rounding either but it's what we use throughout the app (through Ant).

Other than those specific points, I think UI will come to more subjective points, I'm definitely very keen on discussing changes to it, we should only make sure we keep the same elements throughout the website and docs too.

**Comment by lottiecoxon** [comment]
@paolodamico before I get started making these up - are the general concepts good or bad? Be honest as I would like to have these designs done asap 
![Untitled_Artwork-12](https://user-images.githubusercontent.com/65415371/96479761-e8e14000-1228-11eb-85e2-d16f81adb97c.jpg)

**Comment by paolodamico** [comment]
Thanks @lottiecoxon! Hope we can get feedback from everyone else too. Sorry for the weird order.

- For events I like the last one best but I'm still unsure if this is the best option for events.
- For feature flags maybe have two different flags and even a way to distinguish them?
- Toolbar I might consider something less generic that we might reuse as the toolbar "logo" (something in between what we currently have with the PH logo and the generic tool icons).
- I visually like the actions icon. I don't think it's great at conveying what actions are, but also I cannot thing of a better alternative. 
- I like the concept for retention!

**Comment by yakkomajuri** [comment]
As a general rule of thumb, I'm in favor of keeping these as simple as possible. 

That means e.g. for the retention magnet we don't need the little bolt, maybe just the magnet.

Everything looks good to me with the exception of events (too crowded), and maybe the toolbar. The toolbar incites a feel of "settings" to me, which is not something I want to click.

**Comment by jamesefhawkins** [comment]
* I think the last events one looks too much like the way a database is often represented. It's close but not quite the same thing. Sorry that I'm struggling to come up with an alternative concept.
* Toolbar looks like a settings page
* Agree with comments above elsewhere

**Comment by paolodamico** [comment]
Ton of updates on this, opening for another round of feedback.

- Please review fully updated design [on Figma](https://www.figma.com/file/gtW3o3VKYvf04ws8cZM3Nr/PH-Retention-optimisations?node-id=635%3A28).
- https://github.com/PostHog/posthog.com/pull/501 has been updated too with the new proposed navigation items (particularly based on #1828, #1562, #1875). Please note **the design does NOT reflect the actual items for the sidebar**. As the design will not be kept up to date, it's better to keep this info in our docs. Please share any feedback related to the actual items on the referenced issue.
- Icons are placeholders too while we have the finalized versions (see discussion above).

### Changelog
- Addresses feedback (from this thread) from @jamesefhawkins & @lottiecoxon.
- UI style changed significantly based on input from @Twixes & @EDsCODE to be more on brand (and hopefully be more _sleek and retro._). Huge shout out here to the amazing input from both, super helpful!
- Sidebar now reflects options from #1898 where we now have 3 versions ([large screen](https://www.figma.com/file/gtW3o3VKYvf04ws8cZM3Nr/PH-Retention-optimisations?node-id=1067%3A0) for ~15" screens & above, [desktop](https://www.figma.com/file/gtW3o3VKYvf04ws8cZM3Nr/PH-Retention-optimisations?node-id=635%3A28) for tablets & computers up to 15", and [mobile](https://www.figma.com/file/gtW3o3VKYvf04ws8cZM3Nr/PH-Retention-optimisations?node-id=784%3A43). _Note the breakpoints are vaguely defined here pending tests on actual devices._

Sneak peak 👇
![sneak-peak](https://user-images.githubusercontent.com/5864173/96544172-ebca4800-129d-11eb-8fd0-01dee1c242cc.jpg)


### Open discussion
- I want to push back a bit on the suggestion of having the company name be the main element on the top bar (as shown below), based on the following: a) it's more intuitive to have the name at the top (e.g. Google/Gmail, Segment, Stripe, ...); b) multiple orgs won't be applicable to most users; c) at this point it'll be hard to get users to upload the logo for their company and using a placeholder seems a bit confusing. POV @jamesefhawkins ?
<img width="300" alt="" src="https://user-images.githubusercontent.com/5864173/96465531-f8f52180-1220-11eb-9c2f-0cd1acb09064.png">

**Comment by lottiecoxon** [comment]
Great feedback guys- could someone better explain to me what an event is? is it the data collected from the path that a user takes through the app/product... OR is it instead one click equals one event? I feel that the issue with these icons is my misunderstanding of the terms

**Comment by lottiecoxon** [comment]
Feedback please!

![icon ideas](https://user-images.githubusercontent.com/65415371/96577870-d49e5100-12c3-11eb-8145-1853e05c012f.jpg)

**Comment by paolodamico** [comment]
- Like retention & feature flags! Maybe feature flags could be crossed instead of together? Not sure.
- If we're going to start using general filled icon style in the sidebar (which I'm 100% for) I'd go with option A for actions.
- For events I think the first one is too convoluted. I like the second option but I get this feeling that it might mean something else already elsewhere

**Comment by lottiecoxon** [comment]
# Thoughts?

Just realised that the magnet could be made smaller and chubbier so I will get on that now. 

<img width="101" alt="Screenshot 2020-10-20 at 16 22 28" src="https://user-images.githubusercontent.com/65415371/96615324-a4b97280-12f0-11eb-9a44-8c171f96c938.png">

## Toolbar icon redesign 

![Untitled_Artwork](https://user-images.githubusercontent.com/65415371/96615188-805d9600-12f0-11eb-9fd9-4b7c273f078f.jpg)

**Comment by lottiecoxon** [comment]
The colours are inspired by the _classic_ PostHog logo, and done in the 8 bit style, personally I think these are **very fun**, but unsure if they are too distracting - happy to do them in a more toned down monotone. Let me know chaps 

![new 8 bit icons](https://user-images.githubusercontent.com/65415371/96770976-9cd0ff80-13d0-11eb-893b-8f4dc67dbb40.jpg)

**Comment by paolodamico** [comment]
They look **so cool** as standalone icons but I'm thinking that more monotone would be better for this use case. Also, I know this is just an experiment but I'm concerned about how this style of icons will look with the rest of the app using Ant's (might just be a matter of trying it out)

**Comment by lottiecoxon** [comment]
How is this more monotone version? or are the jagged edges still too distracting?
<img width="128" alt="Screenshot 2020-10-27 at 12 19 57" src="https://user-images.githubusercontent.com/65415371/97301743-0f6a3100-1850-11eb-9b00-5abf8a5401f8.png">

**Comment by mariusandra** [comment]
I think this is in the "uncanny valley" territory, where it might just feel like it's a bunch of non-retina gifs as icons from 5-10 years ago. So it's retro, but not retro enough to be oldschool cool 🤷

**Comment by lottiecoxon** [comment]
Tough but fair - I love the concept of them but without the colour they just look a little glitchy :'(

**Comment by mariusandra** [comment]
We could definitely try the colorful ones in an app... and if not directly, then behind an [🥚](https://en.wikipedia.org/wiki/Easter_egg_(media)#:~:text=In%20computer%20software%2C%20Easter%20eggs,otherwise%20serious%20piece%20of%20software.). Now we just need any version implemented to see how it feels like in action and iterate from there.

**Comment by yakkomajuri** [comment]
Direct feedback on my end is that I personally am not a fan of the pixelated/8-bit icons. The actual icons themselves I think are good though.

Final comment would be that with this black and white design (not sure if it will stay that way or not), the feature flags remind me of a race. I'd consider having one flag only.

**Comment by jamesefhawkins** [comment]
> remind me of a race

I now can't un-see this, so I agree with this.

> uncanny valley

Agree that the 8bit style in black/white looks like just some low res icons, which is sad so I'd be a thumbs down for this approach. I adore the colourful ones in isolation, but they are certainly unusual in an app - perhaps they'd be a little distracting in a way that'd detract from their usability.

I wonder if we could take inspiration from the [first icons I ever used, which give me warm and fuzzy feelings](https://www.google.com/search?q=windows+95+icons&rlz=1C5CHFA_enGB833GB833&tbm=isch&source=iu&ictx=1&fir=7H62304M7mj7xM%252C9AOHBRhK56GN7M%252C_&vet=1&usg=AI4_-kQgRwSkRiXKtNqmGqROxY-Nk6WaWg&sa=X&ved=2ahUKEwixrbb56NTsAhVfQUEAHSfzALMQ9QF6BAgOEGo&biw=1745&bih=1831#imgrc=7H62304M7mj7xM), so taking what we have above and toning down the color scheme somehow?

Or perhaps making them more similar to the [earliest mass popularity computer systems](https://www.youtube.com/watch?v=x_r0KJKXyj8)?

Both of above *may* just come across as "too styled". The challenge is doing it in a very lightweight, non-distracting way.

@yakkomajuri could you expand on what you don't like about them? Is it the combo of unusual design and colors that is too much, is 8-bit overdone, something else?

**Comment by paolodamico** [comment]
I think that for the sake of moving this along we could use the "modern" ones or not sure @lottiecoxon if you want to try @jamesefhawkins's suggestions first? I'm also down for @mariusandra's suggestion of putting the colorful ones as an Easter egg, we could even test the icons with a handful of users behind a feature flag as they're already done anyways.

**Comment by lottiecoxon** [comment]
I like the fuzzy warm windows icons - but won't they distract too much ? I tried to keep icons as simple as poss in this 8bit style. I would be up for feature flagging both styles (clean vector and 8bit) to see what works best - and if its not very useable -maybe an easter 🥚 could be a fun way to incorporate them/ or put them on some perch as they are cool

**Comment by yakkomajuri** [comment]
I like the idea of the feature flag from @paolodamico.

@jamesefhawkins In my case, it's really just a personal preference. I'm just not a fan of the "pixelated" look - it triggers PTSD with flashbacks to outdated uni software professors are adamant on using.

While many of us associate the look with "gaming golden days" and things, it just reminds me of software I didn't like. 

But again, it's important to note that this is indeed a strong _personal_ bias that I have. I also am a big fan of the spirit - I like that we try to be different instead of looking like every other startup.

**Comment by lottiecoxon** [comment]
# Comparison 

### From original to 8 bit to redesigned vectors 

Thoughts?

<img width="415" alt="Screenshot 2020-10-27 at 14 41 25" src="https://user-images.githubusercontent.com/65415371/97317334-9aecbd80-1862-11eb-86c7-323db33723b9.png">

**Comment by paolodamico** [comment]
Love option 3! For the toolbar, maybe I would remove the squiggly line on the magnifying glass. Maybe do a horizontal funnel icon for "Funnels" (better represents funnels, plus we can keep the vertical version to mean filter).

Any additional thoughts @EDsCODE @jamesefhawkins @Twixes @yakkomajuri  ?

**Comment by EDsCODE** [comment]
I also like the 3rd one! Agree with @paolodamico's comments. 

Small comment: Is the person's icon crooked? (the neckline seems crooked to me) If it's meant to be, then it should be accented/made more prominent. It confused me a bit
